File: backend/src/app/application/dtos/pagination.py
from dataclasses import dataclass
from typing import Generic, TypeVar

T = TypeVar("T")


@dataclass(frozen=True, slots=True)
class PaginationParams:
    offset: int
    limit: int
    sort_by: str
    sort_order: str


@dataclass(frozen=True, slots=True)
class PaginatedResultDto(Generic[T]):
    items: list[T]
    total: int
    offset: int
    limit: int

File: frontend/src/shared/types/pagination.ts
export const LIST_PAGE_SIZE = 20;

export type SortOrder = "asc" | "desc";

export type PaginationParams<TSortField extends string = string> = {
  offset: number;
  limit: number;
  sort_by: TSortField;
  sort_order: SortOrder;
};

export type PaginatedResponse<T> = {
  items: T[];
  total: number;
  offset: number;
  limit: number;
};
